libs/parallelism/algorithms/tests/unit/container_algorithms/test_utils.hpp

```diff
@@ -19,6 +21,46 @@
 #include <vector>
 
 namespace test {
+    ///////////////////////////////////////////////////////////////////////////
+    // Sentinel constructed from an Iterator just for the purpose of the
+    // overloads tests
+    template <typename IterType>
+    struct Sentinel
```

Comment:
Can we make this `sentinel` instead (i.e. snake case; I know it's capitalized in `iterator_traits`... you could change that one as well)?

Reply:
Done for the sentinel in `test_utils`, will do the others including `Iterator` struct in `iter_sent.hpp` in another PR (see #5075)